### ISSUE TITLE
box.lan does not resolve at the command-line (IIAB installed on Debian 10 Buster VM)

### ISSUE THREAD

**Issue by holta** [iiab/iiab#1815]
@jvonau do you have suggestions or ideas?

It has been rebooted several time.  Live at 10.8.0.30 for now:

```
root@box:~# more /etc/iiab/iiab.env
# This is a configuration file for IIAB
# It can be sourced in a shell script or read into an application
IIAB_BASE_PATH=/opt/iiab
IIAB_DIR=/opt/iiab/iiab
IIAB_RELEASE=7.0
IIAB_REVISION=0
IIAB_GATEWAY_ENABLED=False
LAN_IP=172.18.96.1
OS=debian
OS_VER=debian-10
WWWROOT=/library/www/html
STAGE=9
FQDN=box.lan
IIAB_USB_LIB_SHOW_ALL=True
IIAB_WAN_DEVICE="enp0s3"
IIAB_LAN_DEVICE="none"
HOSTAPD_ENABLED=True
HTTPCACHE_ON=False
```

**Comment by holta** [comment]
@jvonau writes:
> quick fix is to edit /etc/hosts...  as /etc/resolv.conf is pointing upstream..

@holta writes:
> What playbook should do this?

**Comment by holta** [comment]
@jvonau has raised concerns about the existence of `ifupdown` on Debian 10 "Buster", wondering if this might break some aspects of `systemd` ?  See these Debian 10 dependencies:.

```
root@box:/home/iiab-admin# apt-cache depends ifupdown
ifupdown
  Depends: libc6
  Depends: lsb-base
  Depends: adduser
  Depends: iproute2
  Breaks: systemd
 |Recommends: isc-dhcp-client
  Recommends: <dhcp-client>
    dhcpcanon
    dhcpcd5
    pump
    isc-dhcp-client
  Suggests: ppp
  Suggests: rdnssd
  Replaces: systemd
```

ASIDE: `ifupdown` also exists on Ubuntu 18.04 (Desktop version) FWIW.  But does not exist on Ubuntu 18.04's Server version:

```
root@box:~# apt -a list ifupdown
Listing... Done
ifupdown/bionic-updates 0.8.17ubuntu1.1 amd64
ifupdown/bionic 0.8.17ubuntu1 amd64
```

**Comment by holta** [comment]
Curiously, I don't believe Raspbian Buster is affected by these Debian 10 Buster issues?

(But @jvonau & others can correct me if I'm wrong!)

**Comment by holta** [comment]
@jvonau is hoping to solve this in coming days — by comparing fresh "OS only" installs of Debian 10 Buster (released yesterday) and Raspbian Buster (released 2019-06-24) to see what is different.

(PR #1829 is only tangentially related, i.e. Debian 10 brings a few subtle networking changes.)

**Comment by jvonau** [comment]
running iiab-network results in box.lan resolving via /etc/hosts.

**Comment by jvonau** [comment]
hostname returns box.lan while ansible is returning localhost.localdomain when running runrole after the install clobbering box.lan:

TASK [0-init : Set iiab_fqdn: box.lan] ***************************************************************
ok: [127.0.0.1]

TASK [0-init : Set FQDN_changed when iiab_fqdn != ansible_fqdn (localhost.localdomain)] **************
ok: [127.0.0.1]

TASK [0-init : Set hostname if FQDN_changed] *********************************************************
included: /opt/iiab/iiab/roles/0-init/tasks/hostname.yml for 127.0.0.1
<snip>
TASK [0-init : Put hostnames "127.0.0.1 localhost.localdomain localhost box box" in /etc/hosts] ******
changed: [127.0.0.1]
<snip>
changed: [127.0.0.1] => (item={u'option': u'FQDN_changed', u'value': True})

Why is ansible returning "ansible_fqdn": "localhost.localdomain",  "ansible_hostname": "box", 
during ansible -m setup -i ansible_hosts localhost --connection=local
and hostname returns box.lan?

**Comment by jvonau** [comment]
answer is ansible is reading the first entry in /etc/hosts for ansible_fqdn so should the first entry be {{ iiab_hostname }}.{{ iiab_domain }}?

**Comment by holta** [comment]
@jvonau fixed this in PR #1847 -- and as a bonus, amazingly this also fixes name resolution of `box.lan` and `box` starting a few seconds after IIAB's installer begins!
